core/src/main/java/org/bitcoinj/base/internal/Buffers.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright by the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bitcoinj.base.internal;
+
+import org.bitcoinj.base.VarInt;
+
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+
+import static org.bitcoinj.base.internal.Preconditions.check;
+import static org.bitcoinj.base.internal.Preconditions.checkArgument;
+import static org.bitcoinj.base.internal.Preconditions.checkState;
+
+public class Buffers {
```

Comment:
A class-level JavaDoc comment could also be a good place to explain that their are VarInt length fields in the buffer.

Reply:
This class could also be named BitcoinBuffers or P2PBuffers since it contains methods specific to that application.

Reply:
> I've just added a class-level JavaDoc, but explaining VarInt there seems a bit odd. After all, VarInt is only used by two methods.

No need to explain VarInt there, I guess, but I would say: "common operations on Bitcoin P2P message buffers" or something like that.

> I'd rather make its intended audience clear by putting it next to BitcoinSerializer and the Message hierarchy.

That makes sense.

Reply:
> common operations on Bitcoin P2P message buffers

Added this wording.